src/app/route/ProductPage/ProductPage.container.js

```diff
@@ -171,14 +188,15 @@ export class ProductPageContainer extends PureComponent {
             this.getRequiredProductOptions(items);
         }
 
-        if (id !== prevId) {
+        if (id !== prevId || (id === prevId && productName !== metaTitle)) {
             const dataSource = this._getDataSource();
             const { updateMetaFromProduct } = this.props;
 
             updateMetaFromProduct(dataSource);
         }
 
-        this._onProductUpdate();
+        const updateHeader = navName === PDP && navTitle !== prevNavTitle;
```

Comment:
What is this check? Add comment please.

Reply:
This check checks if header needs to be updated to reduce update calls count. Without it popups breaks.